.gitignore

```diff
@@ -35,9 +35,16 @@ TAGS
 # C-Lion / IDEA files
 .idea/**
 
-# Python bytecodes
-__pycache__
+# Python
 *.pyc
+*.egg/
+*.eggs/
+*.egg-info/
+.cache/
+.mypy_cache/
+__pycache__/
+build/
+dist/
```

Comment:
Where does this one come from?

Reply:
Pigweed

Reply:
how about filing an issue with pigweed?  is this really necessary?

Reply:
I mean, the particular list of ignore pattern comes from (is copied from) pigweed. This is more about ignoring the right set of patterns in the rest of CHIP repo.

Reply:
Does that mean we can remove build and dist, these aren't things in CHIP are they?

Reply:
Can we add them as needed? I noticed the .egg-info already and added it in #3291.